### PR TITLE
[front] - chore(mentions): improve mention suggestions

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR improves the mention suggestion dropdown to better prioritize relevant users and agents in conversations. 

The main changes include: prioritizing conversation participants (both users and agents) at the top of the dropdown with a cap of 5 total participants, implementing a 30/70 split between users and agents in suggestion results to ensure user visibility, filtering out the current user from suggestions to prevent self-mentions, promoting the preferred agent (last used) to the first position, and adding visual improvements like a "User" chip to distinguish user mentions and auto-scrolling of selected items.

**References:**
- https://github.com/dust-tt/tasks/issues/5328

## Risk

The logic for mixing and prioritizing suggestions has been significantly refactored in both the frontend (`MentionDropdown.tsx`) and backend (`mention_suggestions.ts`). There's a risk of edge cases in the ordering logic, particularly when handling conversations with few or no participants.

## Tests

- [x] Locally
- [x] front-edge 

## Deploy Plan

Deploy front

### CODE REVIEW

File: front/components/editor/input_bar/MentionDropdown.tsx [hunk 1]
```diff
@@ -19,165 +20,308 @@ import type {
...
+      () => (clientRect ? clientRect() : null),
+      [clientRect]
+    );
+
+    // Fetch suggestions from server using the query.
+    const { suggestions, isLoading } = useMentionSuggestions({
+      workspaceId: owner.sId,
+      conversationId,
+      query,
+      select: { agents: true, users: userMentionsEnabled ?? false },
```
Comment (ElPicador) [inline front/components/editor/input_bar/MentionDropdown.tsx]: you can keep true, there is a check backend side in `front/pages/api/w/[wId]/assistant/conversations/[cId]/mentions/suggestions.ts`

File: front/components/editor/input_bar/MentionDropdown.tsx [hunk 2]
```diff
@@ -19,165 +20,308 @@ import type {
...
+      options: { disabled: !conversationId || !userMentionsEnabled },
+    });
+
+    const { user } = useUser();
+
+    const orderedSuggestions = useMemo(() => {
+      let base = suggestions;
+
+      // Only apply new participant prioritization logic when mentions_v2 is enabled
+      if (userMentionsEnabled) {
```
Comment (ElPicador) [inline front/components/editor/input_bar/MentionDropdown.tsx]: why is this in the front-end and not in the `suggestionsOfMentions` function?

The idea was to keep all the logic in `suggestionsOfMentions`, so we can reuse it elsewhere (like the extension or the agent builder).

**Review by ElPicador — APPROVED** [review]
I would move what is in the frontend to the backend, as per the other comment

File: front/components/assistant/conversation/input_bar/InputBarContainer.tsx [hunk 3]
```diff
@@ -275,6 +275,7 @@ const InputBarContainer = ({
     onUrlDetected: handleUrlDetected,
     owner,
     conversationId,
+    preferredAgentId: selectedAgent?.id ?? null,
```
Comment (ElPicador) [inline front/components/assistant/conversation/input_bar/InputBarContainer.tsx:278]: we agree we can't infer the `selectedAgent` in the backend from the conversation?

Comment (JulesBelveze) [reply front/components/assistant/conversation/input_bar/InputBarContainer.tsx:278]: Not that I'm aware of  :/
